lib/redux/slices/drawer.ts

```diff
@@ -0,0 +1,21 @@
+import { createSlice } from '@reduxjs/toolkit'
+import { DrawerComponentProps } from '@/utils/types/redux'
+
+const initialState: DrawerComponentProps = {
+    drawerDisplay: null,
+    showDrawer: false
+}
+
+export const drawerComponentSlice = createSlice({
+    name: 'drawerComponent',
+    initialState,
+    reducers: {
+        setDrawer: (state, { payload }: { payload: DrawerComponentProps }) => {
+            state.drawerDisplay = payload.drawerDisplay
+            state.showDrawer = payload.showDrawer
+        }
```

Comment:
instead of `setDrawer`, 2 functions.
-----
`showDrawer` that always sets the `show` to true, and sets the content/type to the payload
`closeDrawer` that always sets `show` to false and sets the type/content to initial state/empty string